generation_scripts/abiotic/climate_data_processing_script.py

```diff
@@ -0,0 +1,227 @@
+#
+# description: |
+# This script:
+#   1. Loads a TOML site definition that defines a projected grid
+#   2. Loads a monthly averaged ERA5-Land dataset in NetCDF format (e.g., era5_monthly_2010_2020_maliau.nc)
+#   3. Performs unit conversions (K → °C, m → mm, Pa → kPa, J/m² → W/m²)
+#   4. Adds derived and required variables (relative humidity, constant CO₂, mean annual temperature)
+#   5. Interpolate the gridded ERA5-Land variables to the TOML grid latitude/longitude points (90X90m resolution)
+#   6. Write a VE-style NetCDF that uses x/y/time_index coordinates and variable names
+#   7. Saves processed NetCDF output
+#
+#   The resulting dataset provides climate driver inputs to the Virtual Ecosystem.
+#
+# virtual_ecosystem_module: Abiotic
+#
+# author:
+#   - Lelavathy (adapted from original script)
+#
+# status: wip
+#
+# input_files:
+#   - name: TOML site definition
+#     path: C:\Users\User\OneDrive\Desktop\ve_model\ve_example\generation_scripts\maliau_grid_definition.toml
+#     description: |
+#       Site-specific TOML file defining projected grid cell centre coordinates
+#       (cell_x_centres, cell_y_centres), WGS84 bounds, EPSG code and grid
+#       resolution. Used to build the target 2D grid for interpolation.
+#
+#   - name: ERA5-Land monthly NetCDF (2010–2020)
+#     path: C:\Users\User\OneDrive\Desktop\ve_model\ve_example\era5_monthly_data\2010_2020\ERA5_monthly_2010_2020_Maliau.nc
+#     description: |
+#       Monthly ERA5-Land variables (t2m, d2m, tp, sp, ssrd, u10 etc.)
+#       providing coarse-resolution climate forcing to be regridded to the
+#       Virtual Ecosystem site grid.
+#
+# output_files:
+#   - name: example_climate_from_toml_grid.nc
+#     path: C:\Users\User\OneDrive\Desktop\ve_model\ve_example\data\example_climate_from_toml_grid.nc
+#     description: |
+#       VE-style NetCDF containing variables renamed to Virtual Ecosystem
+#       conventions, with x, y, latitude, longitude, and time_index coordinates.
+#       Ready for ingestion by the VE data loader.
+#
+# package_dependencies:
+#   - numpy
+#   - xarray
+#   - tomllib (Python 3.11+)
+#   - pyproj
+#
+# usage_notes: |
+#   - Uses nearest-neighbour interpolation (xarray.interp with method='nearest').
+#   - EPSG code in the TOML must be valid for pyproj (e.g. 32650).
+#   - Precipitation conversion assumes 30-day months (tp * 1000 * 30).
+#   - Spatial downscaling/topographic correction is not yet implemented.
+# ---
+
+
+import numpy as np
+import tomllib
+import xarray as xr
+from pyproj import Transformer
+
+# ---------------------------
+# File paths (edit here)
+# ---------------------------
+toml_file = (
+    r"C:....ve_data_analysis\abiotic\data_download\site\maliau_grid_definition.toml"
+)
+era5_file = (
+    r"C:.....ve_data_science\data\primary\abiotic\era5_monthly_2010_2020_maliau.nc"
+)
+output_file = (
+    r"C:....ve_data_science\data\derived\abiotic\processed_climate_data_maliau.nc"
+)
```

Comment:
These paths aren't correct - they should not need to be edited and should only be relative to files within the repository